### PR TITLE
integrated full invitation flow into ContactsTab

### PR DESCRIPTION
**Pull request by TedGu**
### What this PR does (required):
-integrated full invitation flow to ContactsTab
-updated `ContactsTab`
-updated `ReferFriend`

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- open ContactsTab in ChatSideBanner, click `+ invite friends` then ReferFriend Dialog opens.

### CODE REVIEW

**Comment by leiwang758** [comment]
### Commit Changes:
- Update UI on `ContactsTab` and `InvitationsTab`
- Integrated backend data and invitation flow into the above tabs UI
- Add API calls and route changes for fetch invitations, accept/reject invitations and fetch contacts.
- Contacts tab pagination and infinite scroll 


### Screenshots / Videos (required):
 - Send invitation
image
- Receive invitation on the other side, can approve or reject
image
- Get contacts based on invitations with back-end integrated infinite scroll and pagination
image

### Current bugs and improvement (required):
- These changes is only for email flow, will work on referral link next
- Right now the user can send invitation to another user even they are already connected, can be improved

**Comment by leiwang758** [comment]
### Commit Changes:
 - Add routes that handles refer link 
image
- Add group chat tab, and connects it to the back end data and `chatboard` (shows the users in the chat and latest message as a preview, a clickon will direct user to the corresponding  `chatboard`, dynamically update chat icon based on the number of users in the chat)
image

File: client/src/components/ChatBox/ChatHeader.tsx [hunk 1]
```diff
@@ -17,16 +18,21 @@ export default function ChatHeader({ handleSwitch, users }: Props): JSX.Element
...
+    ja: 'https://image.flaticon.com/icons/png/128/197/197604.png',
+    ru: 'https://image.flaticon.com/icons/png/128/3909/3909301.png',
+    it: 'https://image.flaticon.com/icons/png/128/3053/3053996.png',
   };
 
+  // const currentChattingUser = {
+  //   image: '/static/images/avatar/currentUser.jpg',
+  //   name: 'Thomas',
+  //   state: 'online',
+  // };
```
Comment (dproc96) [inline client/src/components/ChatBox/ChatHeader.tsx:34]: You can remove comments

**Review by dproc96 — APPROVED** [review]
This looks good, let's go ahead and merge this

**Pull request merged**